microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,209 @@
+Code | Things that grow proportionally to the amount of script, and/or the number of times that functions run..  
+Synthetic | Synthetic nodes don't correspond to anything actually allocated in memory. As far as I can tell, these are used only to distinguish the different kinds of GC roots.
+
+
+## Edges
+
+Similar to the **Nodes** array, edges contains all the individual elements needed to re-construct the edges and we can calculate the number of elements of this array by multiplying `edge_count` * `metadata.edge_fields.length`.
+
+Edges are stored sequentially and to read them properly you need to parse the nodes first, because each node knows how many edges it has. 
+
+To re-construct an edge we need 3 elements of information, and we can use `metadata.edge_types` to understand what each element is. 
+
+For example, if you read the first node and its property edge_count is set to 4, that means that the first 4 triplets of the edges array are the elements needed to from this node to some other nodes, etc.
+
+
+`edge_fields` | Description  
+:- | :- 
+type | See `edge_types[0]` to find out what are the possible types
+name_or_index | This could be a number or a string. If it’s a number, it indicates the index of the strings array where the name is stored.
+to_node | Index in the array of the node that it points to
+
+
+`edge_types` | Description  
+:- | :- 
+Internal | Edges that don't correspond to JS-visible names but are still important. As an example, Function instances have a "context" representing the state of variables that were in scope where the function was defined. There is no way for JS code to directly read the "context" of a function, but it's still a very important link when trying to understand retainers.
+Weak | Weak edges don't keep the target alive, and thus are omitted from the Retainers view. Any object with only weak edges pointing to it can be discarded by the GC.
+Hidden | Just like Internal, except these edges don't have nice names because nobody bothered to write special-case code for them.  They're just numbered in increasing order.
+Shortcut | An easier-to-read representation of some other path. This type is used very little. An example is if you use Function.prototype.bind to create a bound function with some bound arguments, V8 creates a JSBoundFunction, which points to a FixedArray (an internal type), which points to each bound argument. When producing a snapshot, V8 adds a shortcut edge from the bound function directly to each bound argument, bypassing the FixedArray.
+Element | Object properties where the key is a number.
+
+
+<!-- ====================================================================== -->
+## Sample code
+Now that you understand the format of this file you can create your own custom tool to extract data, run an analysis or create a visualization of the memory graph.  You can find sample code under this GitHub repo to get you started. It creates a force directed graph visualization. 
```

Comment:
The link to the github repo is missing here.

Reply:
started a thread to find out where  is the best place to host it publicly.